nfc/doc/type_2_tag.rst

```diff
@@ -0,0 +1,398 @@
+.. _type_2_tag:
+
+Type 2 Tag
+##########
+
```

Comment:
Maybe we should refer to the NFC Forum _Type 2 Tag Technical Specification Version 1.0_ (like with do in case of T4T)? A downside is we'd need to remember to keep this info up-to-date.